packages/sources/accuweather/src/endpoint/current-conditions.ts

```diff
@@ -0,0 +1,262 @@
+    })
+  }
+}
+
+const throwError = (message: string): never => {
+  throw new Error(message)
+}
+
+export const getCurrentConditionsResult = (
+  units: Unit,
+  currentConditionsList: CurrentConditions[],
+): CurrentConditionsResult => {
+  if (!currentConditionsList.length)
+    throw new Error(`Current conditions not found in: ${JSON.stringify(currentConditionsList)}`)
+  if (currentConditionsList.length > 1)
+    throw new Error(
+      `Multiple current conditions found in: ${JSON.stringify(currentConditionsList)}`,
+    )
+
+  const system =
+    unitsConditionKey.get(units) ??
+    throwError(`Unsupported units: ${units}. Supported units are: ${Unit.IMPERIAL},${Unit.METRIC}.`)
+
+  const currentConditions = currentConditionsList[0]
+  const currentConditionsResult: CurrentConditionsResult = {
+    precipitationPast12Hours:
+      currentConditions.PrecipitationSummary?.Past12Hours?.[system as keyof UnitsCondition]?.Value,
+    precipitationPast24Hours:
+      currentConditions.PrecipitationSummary?.Past24Hours?.[system as keyof UnitsCondition]?.Value,
+    precipitationPastHour:
+      currentConditions.PrecipitationSummary?.PastHour?.[system as keyof UnitsCondition]?.Value,
+    precipitationType: precipitationTypeNumber.get(currentConditions.PrecipitationType) as number,
+    pressure: currentConditions.Pressure?.[system as keyof UnitsCondition]?.Value,
+    relativeHumidity: currentConditions.RelativeHumidity,
+    temperature: currentConditions.Temperature?.[system as keyof UnitsCondition]?.Value,
+    timestamp: currentConditions.EpochTime,
+    uvIndex: currentConditions.UVIndex,
+    weatherIcon: currentConditions.WeatherIcon,
+    windDirectionDegrees: currentConditions.Wind?.Direction?.Degrees,
+    windSpeed: currentConditions.Wind?.Speed?.[system as keyof UnitsCondition]?.Value,
+  }
+  Object.entries(currentConditionsResult).forEach(([key, value]) => {
+    value ??
+      throwError(
+        `Missing '${key}' value in: ${JSON.stringify(
+          currentConditionsResult,
+        )}. Current conditions are: ${JSON.stringify(currentConditions)}`,
+      )
+  })
+  // Decimal to integer conversions
+  currentConditionsResult.precipitationPast12Hours = Math.trunc(
+    currentConditionsResult.precipitationPast12Hours * 100,
+  )
+  currentConditionsResult.precipitationPast24Hours = Math.trunc(
+    currentConditionsResult.precipitationPast24Hours * 100,
+  )
+  currentConditionsResult.precipitationPastHour = Math.trunc(
+    currentConditionsResult.precipitationPastHour * 100,
+  )
+  currentConditionsResult.pressure = Math.trunc(currentConditionsResult.pressure * 100)
+  currentConditionsResult.temperature = Math.trunc(currentConditionsResult.temperature * 10)
+  currentConditionsResult.windSpeed = Math.trunc(currentConditionsResult.windSpeed * 10)
+
+  // NB: currently all conditions are expected to be numbers
+  Object.entries(currentConditionsResult).forEach(([key, value]) => {
+    if (Number.isNaN(value))
```

Comment:
Will any of these numbers NOT be integers? If they're all supposed to be ints, we could make this a stronger check (`Number.isInteger(value)`)

Reply:
This is a very good one, thanks!